class-two-factor-core.php

```diff
@@ -825,17 +825,29 @@ public static function login_html( $user, $login_nonce, $redirect_to, $error_msg
 					<?php esc_html_e( 'Having Problems?', 'two-factor' ); ?>
 				</p>
 				<ul>
-					<?php
-					foreach ( $backup_providers as $backup_provider_key => $backup_provider ) :
+				<?php
+					$links = [];
+
+					foreach ( $backup_providers as $backup_provider_key => $backup_provider ) {
 						$backup_link_args['provider'] = $backup_provider_key;
-						?>
-						<li>
-							<a href="<?php echo esc_url( self::login_url( $backup_link_args ) ); ?>">
-								<?php echo esc_html( $backup_provider->get_alternative_provider_label() ); ?>
-							</a>
-						</li>
-					<?php endforeach; ?>
-				</ul>
+						$links[] = sprintf(
+							'<li><a href="$1%s">$2%s</a></li>',
+							esc_url( self::login_url( $backup_link_args ) ),
+							esc_html( $backup_provider->get_alternative_provider_label() )
+						);
+					}
+
+					/*
+					* Allow plugins to add links to the two-factor login form.
+					*/
+					$links = apply_filters( 'two_factor_login_support_links', $links );
+
+					// Echo out the filtered links
+					foreach ( $links as $link ) {
+						echo wp_kses_post( $link );
```

Comment:
**suggestion:** Considering that all of these would be links, could we make the filter run on an array of link hrefs and labels?

```php
$backup_links = [
    'https://example.com' => 'Label',
    // ...
];
```

Reply:
I thought about that, but I can see that some might want to customise the attributes, or not output a link at all.

A generic filter is probably better for that reason.